docs/spire.md

```diff
@@ -0,0 +1,284 @@
+<!--
+---
+linkTitle: "SPIRE"
+weight: 1660
+---
+-->
+# SPIRE TaskRun result attestations
```

Comment:
@geriom You may want to review this document as it will be a whole new page in the reference docs.